### PR TITLE
`Development`: Add git information to header/footer on dev/test server & about page

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Description

This adds a git commit information into the header, when Artemis is run in development mode or on a test server 

image

And also adds git information into the footer (also only in dev mode and on test servers):
image

Commit ID and branch is also added to the about page no matter in which mode artemis is running:
image

### Steps for Testing

1. Start Artemis in dev mode
2. Mouseover the version in the header & have a look at the footer & about page
3. Run artemis on a test server and repeat 2.
4. Run artemis on a prod server, only about page should have information

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/webapp/app/shared/layouts/navbar/navbar.component.ts [hunk 1]
```diff
@@ -177,7 +182,12 @@ export class NavbarComponent implements OnInit, OnDestroy {
         this.profileService.getProfileInfo().subscribe((profileInfo) => {
             if (profileInfo) {
                 this.inProduction = profileInfo.inProduction;
+                this.testServer = profileInfo.testServer == undefined ? false : profileInfo.testServer;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/shared/layouts/navbar/navbar.component.ts:185]:
```suggestion
                this.testServer = profileInfo.testServer ?? false;
```

File: src/main/webapp/i18n/de/git.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "git": {
+        "title": "Git Information",
+        "branch": "Branch",
+        "commit": "Commit ID",
+        "timestamp": "Commit Zeitstempel",
+        "username": "Commit Benutzername"
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/git.json:7]: The German nouns should either contain hyphens (-) or be merged to one word

File: src/main/webapp/i18n/de/git.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "git": {
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/git.json:2]: We normally have artemisApp as the first level in the translation files

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JohannesStoehr — DISMISSED** [review]
Thanks for incorporating my suggestions. Code looks good to me

**Review by pellumbbaboci — DISMISSED** [review]
Tested on ts1 and locally, works as expected and code LGTM. 
There are conflicts with develop branch as recently there were changes in footer components. Can you resolve those conflicts?

**Comment by TheZoker** [comment]
Thanks for the hint! I will rebase the PR

**Review by pellumbbaboci — CHANGES_REQUESTED** [review]
As seen in the picture, footer is showing incorrectly. Can you check again?

footer

**Comment by TheZoker** [comment]
Thanks for the hint, it seems that one `row` class got removed during the footer PR. I'm not sure why the client-tests fail, since the `ngbTooltip` prop of the div was not an issue before, when the test passed. I'll investigate

File: src/main/webapp/app/shared/layouts/footer/footer.component.ts [hunk 4]
```diff
@@ -14,6 +14,12 @@ export class FooterComponent implements OnInit {
     readonly requestChangeUrl = 'https://github.com/ls1intum/Artemis/issues/new/choose';
 
     email: string;
+    gitBranch: string;
+    gitCommitId: string;
+    gitTimestamp: string;
+    gitCommitUser: string;
+    testServer: Boolean;
```
Comment (danielbode) [inline src/main/webapp/app/shared/layouts/footer/footer.component.ts:21]:
```suggestion
    testServer: boolean;
```

File: src/main/webapp/app/shared/layouts/footer/footer.component.ts [hunk 5]
```diff
@@ -14,6 +14,12 @@ export class FooterComponent implements OnInit {
...
+    inProduction: Boolean;
```
Comment (danielbode) [inline src/main/webapp/app/shared/layouts/footer/footer.component.ts:22]:
```suggestion
    inProduction: boolean;
```

Comment (danielbode) [reply src/main/webapp/app/shared/layouts/footer/footer.component.ts:22]: It is recommended to always use primitive boolean type for declarations: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#the-primitives-string-number-and-boolean

File: src/main/webapp/i18n/de/git.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+    "artemisApp": {
+        "git": {
+            "title": "Git-Information",
```
Comment (danielbode) [inline src/main/webapp/i18n/de/git.json:4]:
```suggestion
            "title": "Git-Informationen",
```

**Review by danielbode — CHANGES_REQUESTED** [review]
Nice PR, works as expected. Tested locally and on ts0. I just added some feedback to the code.

It would also be nice to format the timestamp. I think `Nov 28, 2022 20:44:06` looks better than `2022-11-28T20:44:06Z`.

**Review by tobias-lippert — APPROVED** [review]
Tested locally and on ts1. Works as expected. Nice improvement.

**Review by danielbode — APPROVED** [review]

File: src/main/webapp/app/shared/layouts/footer/footer.scss [hunk 7]
```diff
@@ -11,4 +11,13 @@
         font-weight: normal;
         font-size: small;
     }
+
+    .footer-git-wrapper {
+        display: flex;
+        .footer-git {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/shared/layouts/footer/footer.scss:17]: Codacy

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by pellumbbaboci — APPROVED** [review]
LGTM

**Review by rriyaldhi — APPROVED** [review]
Tested on TS1 and works as expected.

**Review by evgeny1984 — APPROVED** [review]
Tested on the ts1. Everything worked as expected on different screen sizes. The code also looks good.

**Review by danielbode — APPROVED** [review]
Reapprove after adding tests.

**Review by pellumbbaboci — APPROVED** [review]
REAPROVE

**Pull request merged**